tapdb/supply_ignore_checker.go

```diff
@@ -0,0 +1,224 @@
+// is used here, but the check should be tested against a proof result, or
+// produced output.
+func (c *CachingIgnoreChecker) IsIgnored(ctx context.Context,
+	prevID proof.AssetPoint) lfn.Result[bool] {
+
+	c.Lock()
+	defer c.Unlock()
+
+	log.Tracef("Checking if asset point %s is ignored", prevID)
+
+	// Let's do a direct lookup first to see if we already know the asset is
+	// ignored.
+	if c.ignoredAssetPoints.Contains(prevID) {
+		log.Tracef("Asset point %s is already known to be ignored",
+			prevID)
+		return lfn.Ok(true)
+	}
+
+	// If it's not known to be ignored, we need to look up the group key for
+	// the asset ID associated with the previous ID. This will help us to
+	// decide whether it's a non-grouped asset (which is never ignored) or
+	// to fetch the latest ignore tuples for the asset group.
+	groupKey, groupKeyKnown := c.groupKeyLookup[prevID.ID]
+	if !groupKeyKnown {
+		log.Tracef("Asset point %s is not known, querying group "+
+			"key for asset ID %s", prevID, prevID.ID)
+
+		group, err := c.cfg.GroupQuery.QueryAssetGroupByID(
+			ctx, prevID.ID,
+		)
+		switch {
+		// We found the group, and it has a valid group key, we can add
+		// it to our lookup map.
+		case err == nil && group.GroupKey != nil:
+			groupKey = asset.ToSerialized(
+				&group.GroupKey.GroupPubKey,
+			)
+			c.groupKeyLookup[prevID.ID] = groupKey
+
+			log.Tracef("Found group key %s for asset ID %s",
+				groupKey, prevID.ID)
+
+		// If the asset is not a grouped asset, we mark it as such in
+		// our lookup map.
+		case err == nil && group.GroupKey == nil:
+			groupKey = nonGroupedAsset
+			c.groupKeyLookup[prevID.ID] = groupKey
+
+			log.Tracef("Asset ID %s is not grouped, using "+
+				"non-grouped asset key", prevID.ID)
+
+		case err != nil:
+			// If something went wrong while querying the asset
+			// group, we return an error.
+			if !errors.Is(err, address.ErrAssetGroupUnknown) {
+				return lfn.Errf[bool]("unable to query asset "+
+					"group: %w", err)
+			}
+
+			// If we get here, it means we are not aware of this
+			// asset group, so we can assume it is not ignored.
+			return lfn.Ok(false)
+		}
+	}
+
+	// We now have the group key, so we can check if the asset is a
+	// non-grouped asset, which means it is not ignored (as that
+	// functionality is only supported for grouped assets).
+	if groupKey == nonGroupedAsset {
+		log.Tracef("Asset point %s is a non-grouped asset, ignore "+
+			"functionality not available", prevID)
+		return lfn.Ok(false)
+	}
+
+	groupPubKey, err := groupKey.ToPubKey()
+	if err != nil {
+		return lfn.Errf[bool]("unable to decode group key: %w", err)
+	}
+
+	// If we have a valid group key, we can fetch the latest ignore
+	// tuples for the asset group based on the best known height for the
+	// group (or 0 if we haven't encountered this group yet).
+	specifier := asset.NewSpecifierFromGroupKey(*groupPubKey)
+	bestHeight := c.bestHeightLookup[groupKey]
+	leaves, err := c.cfg.Store.FetchSupplyLeavesByType(
```

Comment:
One alternative here would be doing a more precise query just that that particular ignore point. On the other ahdn wit hthis approach, we do the broad look up once, then just cache it from there on.